app/controllers/users_controller.rb

```diff
@@ -45,6 +46,42 @@ def index
     end
   end
 
+  def show
+    user = User.find(params.require(:id))
+
+    render status: :ok, json: user.as_json(
+      include: %w[roles],
+      methods: %w[special_account_competitions],
+      private_attributes: %w[email],
+    )
+  end
+
+  def merge
+    to_user = User.find(params.require(:toUserId))
+    from_user = User.find(params.require(:fromUserId))
+
+    return render status: :bad_request, json: { error: "Cannot merge user with itself" } if to_user.id == from_user.id
+
+    if to_user.name != from_user.name ||
+       to_user.country_iso2 != from_user.country_iso2 ||
+       to_user.gender != from_user.gender ||
+       to_user.dob != from_user.dob
+      return render status: :bad_request, json: { error: "Cannot merge users with different details" }
+    end
+
+    if !current_user.results_team? && (to_user.special_account? || from_user.special_account?)
+      return render status: :bad_request,
+                    json: { error: 'One of the account is a special account, please contact WRT to merge them.' }
+    end
+
+    return render status: :bad_request, json: { error: "Cannot merge users with both having a WCA ID" } if to_user.wca_id.present? && from_user.wca_id.present?
+
+    ActiveRecord::Base.transaction do
+      from_user.transfer_data_to(to_user)
+    end
```

Comment:
A transaction with one statement looks a bit silly. If the changes within `transfer_data_to` should be atomic, then wrap a transaction "internally" within the method body of `def transfer_data_to`.